src/main/java/com/cspark/nextstep/step3/domain/CarRace.java

```diff
@@ -0,0 +1,57 @@
+package com.cspark.nextstep.step3.domain;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class CarRace {
+
+  private final int roundCount;
+  private final List<Car> cars;
+
+  private int currentCount = 0;
+
+  public CarRace(int roundCount, String[] carNames, Function<Dice, Boolean> rule) {
+    this.roundCount = roundCount;
+    this.cars = makeCars(carNames, rule);
+  }
+
+  private List<Car> makeCars(String[] carNames, Function<Dice, Boolean> rule) {
+    List<Car> cars = new ArrayList<>();
+
+    for (String carName : carNames) {
+      cars.add(new Car(carName, rule));
+    }
+
+    return cars;
+  }
+
+  public List<Scorecard> race(Dice dice) {
+    currentCount++;
+    for (Car car : cars) {
+      car.drive(dice);
+    }
+    return scorecard();
+  }
+
+  private List<Scorecard> scorecard() {
+    return cars.stream()
+        .map(d -> Scorecard.of(d))
```

Comment:
d라는 네이밍은 무엇을 뜻하는지 이해하기 힘든 것 같아요 ㅠㅠ
```suggestion
        .map(car -> Scorecard.of(car))
```